### PR TITLE
Initial support for filelist

### LINKED ISSUE #37: Global List of videos and other files
ipcRenderer listener `'return-videos'` and `'return-pbsgen'` in `video.tsx `are buggy, sometime they just don't get registered properly.

Not sure what the issue is, but one possible solution is to keep a global list of videos and other files, this also would be really nice to have when we display a list of files on the profile page.

### PR DESCRIPTION
**Pull request by daniel-luan**
Add initial support for filelist

closes #37

### CODE REVIEW

File: app/features/video/Video.tsx [hunk 1]
```diff
@@ -158,42 +161,8 @@ function VideoPlayer(props: VideoPlayerProps): JSX.Element {
...
-    return () => {
-      ipcRenderer.removeAllListeners('return-videos');
-      ipcRenderer.removeAllListeners('return-pbsgen');
-    };
-  }, [files, course, dispatch]);
-
-  useEffect(() => {
-    ipcRenderer.send('get-videos', wkspace, course);
-  }, [course, wkspace]);
+  const files = FileList.getVideoFiles(course);
```
Comment (daniel-luan) [inline app/features/video/Video.tsx:165]: Not sure how to get react to rerender when filelist updates.
Everything else works, will look into this.

Comment (kevinli23) [reply app/features/video/Video.tsx:165]: you can make a custom hook which would force it to rerender i think

File: app/features/filelist/filelist.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { ipcRenderer } from 'electron';
+
+import { VideoData } from '../../interfaces';
+
+interface CourseData {
+  video: VideoData[];
+  assign: VideoData[];
+  pdf: VideoData[];
+}
```
Comment (kevinli23) [inline app/features/filelist/filelist.ts:9]: Should they all be VideoData[]?

**Review by kevinli23 — CHANGES_REQUESTED** [review]
Prob merge master and fix the merge conflicts

**Comment by daniel-luan** [comment]
I added the callback stuff, and it seems to be working. But react is being weird, idk why it's not rerendering.

**Comment by kevinli23** [comment]
Weird, i think it is fine for now, we can just put it as separate issue for now

**Comment by daniel-luan** [comment]
Okay I fixed the merge conflict, everything should be working beside the proper rerendering.

**Review by kevinli23 — APPROVED** [review]

**Comment by kevinli23** [comment]
yup looks good

**Pull request merged**